lib/Sema/CSSimplify.cpp

```diff
@@ -5632,6 +5632,60 @@ ConstraintSystem::simplifyApplicableFnConstraint(
     return simplified;
   }
 
+  // Handle applications of types with call methods.
+  if (desugar2->mayHaveMembers()) {
+    auto &ctx = getASTContext();
+    // Get all call methods of the nominal type.
+    // TODO: Consider caching?
+    SmallVector<FuncDecl *, 4> callMethods;
```

Comment:
Todo: consider caching the `func call` methods for types?

Such caching of methods is done for [`@dynamicCallable`](https://github.com/apple/swift/blob/master/lib/Sema/ConstraintSystem.h#L1136). There's also a [cache for `@dynamicMemberLookup` types](https://github.com/apple/swift/blob/master/lib/Sema/ConstraintSystem.h#L1139), but it doesn't store methods.